### PR TITLE
SONAR-10452 Empty "On new code" sections on Measures page

### PR DESCRIPTION
**Pull request by pascal-mugnier-sonarsource**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 1]
```diff
@@ -77,10 +77,18 @@ export default class DomainFacet extends React.PureComponent {
 
   renderItemsFacet = () => {
     const { domain, selected } = this.props;
+
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: remove this change

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 2]
```diff
@@ -77,10 +77,18 @@ export default class DomainFacet extends React.PureComponent {
...
     const items = addMeasureCategories(domain.name, filterMeasures(domain.measures));
     const hasCategories = items.some(item => typeof item === 'string');
     const translateMetric = hasCategories ? getLocalizedCategoryMetricName : getLocalizedMetricName;
-    const sortedItems = sortMeasures(domain.name, items);
+    let sortedItems = sortMeasures(domain.name, items);
+
+    sortedItems = sortedItems.filter((item, id, array) => {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: remove `array` and use `sortedItems`

Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: please replace `id` with `index`

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/__tests__/DomainFacet-test.js [hunk 3]
```diff
@@ -69,3 +69,32 @@ it('should render closed', () => {
...
+          domain: 'Reliability'
+        },
+        value: '5',
+        periods: [{ index: 1, value: '5' }],
+        leak: '5'
+      }
+    ]
+  };
+
+  const props = {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/__tests__/DomainFacet-test.js]: inline this

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/__tests__/DomainFacet-test.js [hunk 4]
```diff
@@ -69,3 +69,32 @@ it('should render closed', () => {
   const wrapper = shallow(<DomainFacet {...PROPS} open={false} />);
   expect(wrapper.find('FacetItemsList')).toHaveLength(0);
 });
+
+it('should not display subtitles of new measures if there is none', () => {
+  const domain = {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/__tests__/DomainFacet-test.js]: try to reduce amount of data, i.e. keep only mandatory properties

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 5]
```diff
@@ -77,10 +77,18 @@ export default class DomainFacet extends React.PureComponent {
...
+      return (
+        typeof item !== 'string' || (array.length > id - 1 && typeof array[id + 1] !== 'string')
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: it won't work if the subtitle is the last one in the list

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 6]
```diff
@@ -80,11 +80,17 @@ export default class DomainFacet extends React.PureComponent {
     const items = addMeasureCategories(domain.name, filterMeasures(domain.measures));
     const hasCategories = items.some(item => typeof item === 'string');
     const translateMetric = hasCategories ? getLocalizedCategoryMetricName : getLocalizedMetricName;
-    const sortedItems = sortMeasures(domain.name, items);
+    let sortedItems = sortMeasures(domain.name, items);
+
+    sortedItems = sortedItems.filter((item, index) => {
+      if (item !== 'string') return true;
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: Code Smell Code Smell: Add curly braces around the nested statement(s) in this "if" block. ([javascript:CurlyBraces](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ACurlyBraces))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fpm%2Ffix-sonar-10452&issues=AWH60zaka5Ltb7kATuXC&open=AWH60zaka5Ltb7kATuXC)

[ISSUE_KEY]: AWH60zaka5Ltb7kATuXC

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 7]
```diff
@@ -67,9 +67,10 @@ export default class DomainFacet extends React.PureComponent {
     const { domain, selected } = this.props;
     const measureSelected = domain.measures.find(measure => measure.metric.key === selected);
     const overviewSelected = domain.name === selected && hasBubbleChart(domain.name);
-    return measureSelected
-      ? [getLocalizedMetricName(measureSelected.metric)]
-      : overviewSelected ? [translate('component_measures.domain_overview')] : [];
+    if (measureSelected) {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: Code Smell Code Smell: Add curly braces around the nested statement(s) in this "if" block. ([javascript:CurlyBraces](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ACurlyBraces))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fpm%2Ffix-sonar-10452&issues=AWH6-c6ma5Ltb7kATvqy&open=AWH6-c6ma5Ltb7kATvqy)

[ISSUE_KEY]: AWH6-c6ma5Ltb7kATvqy

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 8]
```diff
@@ -80,11 +81,19 @@ export default class DomainFacet extends React.PureComponent {
     const items = addMeasureCategories(domain.name, filterMeasures(domain.measures));
     const hasCategories = items.some(item => typeof item === 'string');
     const translateMetric = hasCategories ? getLocalizedCategoryMetricName : getLocalizedMetricName;
-    const sortedItems = sortMeasures(domain.name, items);
+    let sortedItems = sortMeasures(domain.name, items);
+
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: Code Smell Code Smell: Add curly braces around the nested statement(s) in this "if" block. ([javascript:CurlyBraces](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3ACurlyBraces))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fpm%2Ffix-sonar-10452&issues=AWH6-c6ma5Ltb7kATvqz&open=AWH6-c6ma5Ltb7kATvqz)

[ISSUE_KEY]: AWH6-c6ma5Ltb7kATvqz

File: server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js [hunk 9]
```diff
@@ -80,11 +81,19 @@ export default class DomainFacet extends React.PureComponent {
...
+    sortedItems = sortedItems.filter((item, index) => {
+      if (typeof item !== 'string') {
+        return true;
+      }
+      return sortedItems.length === index - 1 || typeof sortedItems[index + 1] !== 'string';
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: `index - 1` is wrong

Comment (stas-vilchik) [reply server/sonar-web/src/main/js/apps/component-measures/sidebar/DomainFacet.js]: and please add test for this case

**Pull request merged**